concepts/0094-cross-domain-messaging/README.md

```diff
@@ -0,0 +1,302 @@
+
+An Agent Message defines the format of messages processed by Agents. Details about the general form of Agent Messages can be found in the [Agent Messages](../0020-message-types/README.md) RFC.
+
+This RFC specifies (below) the "Forward" message type, a part of the "Routing" family of Agent Messages.
+
+### DID, DIDDoc and Routing
+
+A DID owned by the Receiver is resolvable by the Sender as a DIDDoc using either a Public Ledger or using the [did:peer spec](https://dhh1128.github.io/peer-did-method-spec/index.html). The related [DIDcomm DIDDoc Conventions](../../features/0067-didcomm-diddoc-conventions/README.md) RFC defines the required contents of a DIDDoc created by the receiving entity. Notably, the DIDDoc given to the Sender by the Receiver specifies the required routing of the message through an optional set of [mediators](../0046-mediators-and-relays/README.md).
+
+### Cross Domain Interoperability
+
+A key goal for interoperability is that we want other domains to know just enough about the configuration of a domain to which they are delivering a message, but no more. The following walks through those minimum requirements.
+
+#### Required: The DID and DIDDoc
+
+As noted above, the Sender of an Agent to Agent Message has the DID of the Receiver, and knows the key(s) from the DIDDoc to use for the Receiver's Agent(s).
+
+> Example: Alice wants to send a message from her phone (1) to Bob's phone (4). She has Bob's B:did@A:B, the DID/DIDDoc Bob created and gave to Alice to use for their relationship. Alice created A:did@A:B and gave that to Bob, but we don't need to use that in this example. The content of the DIDDoc for B:did@A:B is presented above.
+
+#### Required: End-to-End encryption of the Agent Message
+
+The Agent Message from the Sender SHOULD be hidden from all Agents other than the Receiver. Thus, it SHOULD be encrypted with the public key of the Receiver. Based on our assumptions, the Sender can get the public key of the Receiver agent because they know the DID#keyname string, can resolve the DID to the DIDDoc and find the public key associated with DID#keyname in the DIDDoc. In our example above, that is the key associated with "did:sov:1234abcd#4".
+
+Most Sender-to-Receiver messages will be sent between parties that have shared pairwise DIDs. When that is true, the Sender will (usually) AuthCrypt the message. If that is not the case, or for some other reason the Sender does not want to AuthCrypt the message, AnonCrypt will be used. In either case, the Indy-SDK `pack()` function handles the encryption.
+
+If there are mediators specified in the DID service endpoint for the Receiver agent, the Sender must wrap the message for the Receiver in a 'Forward' message for each mediator. It is assumed that the Receiver can determine the from `did` based on the `to` DID using their pairwise relationship.
+
+```json
+{
+  "@type" : "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/routing/1.0/forward",
+  "to"   : "verkey of last mediator",
```

Comment:
@swcurran according to the text preceding this example, the `to` field should have the did fragment of the Receiver's key

Reply:
Thanks - reverted. I confused myself on that.